homeassistant/components/unifi/controller.py

```diff
@@ -395,15 +394,27 @@ def shutdown(self, event: Event) -> None:
 
         Used as an argument to EventBus.async_listen_once.
         """
-        self.api.stop_websocket()
+        if self.ws_task is not None:
+            self.ws_task.cancel()
 
     async def async_reset(self) -> bool:
         """Reset this controller to default state.
 
         Will cancel any scheduled setup retry and will unload
         the config entry.
         """
-        self.api.stop_websocket()
+        if self.ws_task is not None:
+            self.ws_task.cancel()
+
+            _, pending = await asyncio.wait([self.ws_task], timeout=10)
+
+            if pending:
+                LOGGER.warning(
+                    "Unloading %s (%s) config entry. Task %s did not complete in time",
+                    self.config_entry.title,
+                    self.config_entry.domain,
+                    self.ws_task,
```

Comment:
Feels like overkill loving here. Not sure you should timeout at all. Caller of service can handle the timeout.

Reply:
Absolutely, but then I can see if that part is relevant over time and can rework it later on. Right now I want to get it in before beta cut off :)

Happy for now?